mutiny-core/src/scb/mod.rs

```diff
@@ -0,0 +1,664 @@
+type Aes256CbcDec = Decryptor<Aes256>;
+
+pub const SCB_ENCRYPTION_KEY_DERIVATION_PATH: &str = "m/444'/444'/444'";
+
+/// A static channel backup is a backup for the channels for a given node.
+/// These are backups of the channel monitors, which store the necessary
+/// information to recover the channel in case of a failure.
+#[derive(Default, PartialEq, Eq, Clone)]
+pub struct StaticChannelBackup {
+    /// Map of the channel outpoint to the channel monitor
+    /// This is a Vec<u8> because we can't implement Readable for ChannelMonitor
+    /// without having a KeysManager, which we don't have here.
+    pub(crate) monitors: HashMap<OutPoint, Vec<u8>>,
+}
+
+impl Writeable for StaticChannelBackup {
+    fn write<W: Writer>(&self, writer: &mut W) -> Result<(), lightning::io::Error> {
+        let len: u32 = self.monitors.len() as u32;
+        writer.write_all(&len.to_be_bytes())?;
+        for (outpoint, monitor) in self.monitors.iter() {
+            writer.write_all(&outpoint.txid[..])?;
+            writer.write_all(&outpoint.vout.to_be_bytes())?;
+            let mon_len: u32 = monitor.len() as u32;
+            writer.write_all(&mon_len.to_be_bytes())?;
+            writer.write_all(monitor)?;
+        }
+        Ok(())
+    }
+}
+
+impl Readable for StaticChannelBackup {
+    fn read<R: Read>(reader: &mut R) -> Result<Self, DecodeError> {
+        let len: u32 = Readable::read(reader)?;
+        let mut monitors = HashMap::new();
+        for _ in 0..len {
+            let mut txid = [0u8; 32];
+            reader.read_exact(&mut txid)?;
+            let vout: u32 = Readable::read(reader)?;
+            let outpoint = OutPoint {
+                txid: bitcoin::Txid::from_slice(&txid).expect("txid is 32 bytes"),
+                vout,
+            };
+            let mon_len: u32 = Readable::read(reader)?;
+            let mut monitor = vec![0u8; mon_len as usize];
+            reader.read_exact(&mut monitor)?;
+            monitors.insert(outpoint, monitor);
+        }
+
+        Ok(Self { monitors })
+    }
+}
+
+/// A static channel backup storage contains the static channel backups
+/// for all of the node manager's nodes.
+///
+/// This also has the NodeStorage, which contains the the necessary
+/// information to recover the node manager's nodes.
+#[derive(Default, PartialEq, Eq, Clone)]
+pub struct StaticChannelBackupStorage {
+    pub(crate) backups: HashMap<PublicKey, (NodeIndex, StaticChannelBackup)>,
+    pub(crate) peer_connections: HashMap<PublicKey, String>,
+}
+
+impl StaticChannelBackupStorage {
+    pub(crate) fn encrypt(&self, secret_key: &SecretKey) -> EncryptedSCB {
+        let bytes = self.encode();
+        let iv: [u8; 16] = secp256k1::rand::random();
+
+        let cipher = Aes256CbcEnc::new(&secret_key.secret_bytes().into(), &iv.into());
+        let encrypted_scb: Vec<u8> = cipher.encrypt_padded_vec_mut::<Pkcs7>(&bytes);
+
+        EncryptedSCB { encrypted_scb, iv }
+    }
```

Comment:
We have encrypt and decrypt methods already, should we consolidate them in any way? I guess the only difference between the two should just be one is encrypted with a password or one is encrypted with the seed bytes.

Reply:
Yeah I tried but wasn't really clean, most of the code in our current encrypt and decrypt functions are turning the password into a key. I'll try to take a stab at it again

Reply:
We might switch to https://crates.io/crates/fedimint-aead so will ignore for that for now.

Reply:
> Is the encryption / decryption path hit right now? We can start playing with it but not ship the feature until it's ready. Otherwise we can't deploy with one encryption method and then switch it later easily.

We don't currently use our encrypt/decrypt for the storage because that isn't exposed. This PR adds a different encrypt/decrypt method (just copy pasted from rust-nostr) since we are using keys, not passwords. I think this encryption won't change but we might change how our encrypted local storage works because it is just something I copy pasted from some chineese code

Reply:
Yeah good points, you're right that the local password based encryption method wouldn't be used here at all, so nothing wrong with leaving this for good.